bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java

```diff
@@ -724,15 +717,12 @@ int runCmd(CommandLine cmdLine) throws Exception {
 
         @Override
         public int runCmd(CommandLine cmdLine) throws Exception {
-            ZooKeeper zk = null;
             LedgerManagerFactory mFactory = null;
             LedgerManager m = null;
             try {
-                zk = ZooKeeperClient.newBuilder()
-                        .connectString(bkConf.getZkServers())
-                        .sessionTimeoutMs(bkConf.getZkTimeout())
-                        .build();
-                mFactory = LedgerManagerFactory.newLedgerManagerFactory(bkConf, zk);
+                mFactory = LedgerManagerFactory.newLedgerManagerFactory(
```

Comment:
I think we are lacking some close().
You can considering even adding a close() method to LedgerManagerFactory and make it AutoCloseable in order to use try-with-resources

Reply:
Yes, have thought do the changes. will do it.